packages/builders/src/cypress/cypress.builder.ts

```diff
@@ -0,0 +1,182 @@
+ * @example:
+```
+ "my-app-e2e": {
+    "root": "apps/my-app-e2e/",
+    "projectType": "application",
+    "architect": {
+      "e2e": {
+        "builder": "@nrwl/builders:cypress",
+        "options": {
+          "cypressConfig": "apps/my-app-e2e/cypress.json",
+          "tsConfig": "apps/my-app-e2e/tsconfig.json",
+          "devServerTarget": "my-app:serve"
+      },
+      "configurations": {
+        "production": {
+          "devServerTarget": "my-app:serve:production"
+        }
+      }
+      }
+    }
+ }
+```
+ *
+ */
+export default class CypressBuilder implements Builder<CypressBuilderOptions> {
+  private computedCypressBaseUrl: string;
+
+  constructor(public context: BuilderContext) {}
+
+  /**
+   * @whatItDoes This is the starting point of the builder.
+   * @param builderConfig
+   */
+  run(
+    builderConfig: BuilderConfiguration<CypressBuilderOptions>
+  ): Observable<BuildEvent> {
+    const options = builderConfig.options;
+
+    return this.compileTypescriptFiles(options.tsConfig).pipe(
+      concatMap(() => this.startDevServer(options)),
+      concatMap(() =>
+        this.initCypress(options.cypressConfig, options.headless)
+      ),
+      catchError(error => {
+        throw new Error(error);
+      })
+    );
+  }
+
+  /**
+   * @whatItDoes Compile typescript spec files to be able to run Cypress.
+   * The compilation is done via executing the `tsc` command line/
+   * @param tsConfigPath
+   */
+  private compileTypescriptFiles(tsConfigPath: string): Observable<BuildEvent> {
+    return Observable.create(subscriber => {
+      try {
+        const output = spawn(
+          `${path.resolve(this.context.workspace.root)}/node_modules/.bin/tsc`,
+          ['-p', tsConfigPath],
+          { stdio: [0, 1, 2] }
+        );
+        output.on('exit', code => subscriber.next({ success: code === 0 }));
+      } catch (error) {
+        subscriber.error(
+          new Error(`Could not compile Typescript files: \n ${error}`)
+        );
+      }
+    });
+  }
+
+  /**
+   * @whatItDoes Initialize the Cypress test runner with the provided project configuration.
+   * If `headless` is `false`: open the Cypress application, the user will
+   * be able to interact directly with the application.
+   * If `headless` is `true`: Cypress will run in headless mode and will
+   * provide directly the results in the console output.
+   * @param cypressConfig
+   * @param headless
+   */
+  private initCypress(
+    cypressConfig: string,
+    headless: boolean
+  ): Observable<BuildEvent> {
+    // Cypress expects the folder where a `cypress.json` is present
+    const projectFolderPath = path.dirname(cypressConfig);
+    const options = {
+      config: { baseUrl: this.computedCypressBaseUrl },
+      project: projectFolderPath
+    };
+    return fromPromise(
+      headless ? Cypress.run(options) : Cypress.open(options)
+    ).pipe(tap(() => process.exit()), mapTo({ success: true }));
```

Comment:
Are you sure?
As i understand, the build process is not tied to the result of the tests.
So wether or not the tests are passing, the _BuildEvent_ is completed because it handles:
- compiling files from typescript into javascript
- copying the fixtures data
- starting Cypress

See Protractor build, it is not failing if the tests are failing, it is failing if you have a typo, a file missing or something that is relate to the build mechanism itself.

Do i make sense?